application/dash_application/utility/plot_builder.py

```diff
@@ -827,15 +846,29 @@ def __init__(
         self.bar_positive = bar_positive
         self.bar_negative = bar_negative
         self.log_y = log_y
+        self.mode = mode
 
     def traces(self):
+        self.display_x = None
+        self.x_fn = None
+        if self.x_col is None:
+            if self.mode == Mode.IUS:
+                self.x_fn = lambda d: d["SampleNameExtra"]
+                self.display_x = lambda d: d[PINERY_COL.SampleName]
+            elif mode == Mode.MERGED:
+                self.x_fn = lambda d: d[ml_col]
+                self.display_x = lambda d: d[ml_col]
+        else:
+            self.display_x = lambda d: x_fn(d)
+
```

Comment:
Would is make sense to move above code to `_generate_traces` (rather than `generate`), to avoid this code duplication?

Reply:
It is this way because the 'categoryarray' for the x-axis needs access to the correctly set x_fn.

Reply:
Ok, my main fear was that `generate` and here could become divergent. I think that `generate` won't be called anymore once we switch over to subplots, so this is fine.